scripts/pylib/twister/twisterlib.py

```diff
@@ -3388,6 +3385,13 @@ def load_from_file(self, file, filter_status=[], filter_platform=[]):
                         tfilter,
                         self.fixtures
                     )
+                    instance.status = ts['status']
```

Comment:
why do we use the last status to set current instance status, maybe use it's initialized status 'None' it's better, because this is a new instance, but for instance.results, we should initialize all sub cases' status like what we are doing in apply_filter, this is useful for due to some reasons, this test instance didn't get executed but we need do fill_results_by_status.
```
 for t in tc.cases:                                                                                                                  
                    instance.results[t] = None
```

Reply:
Yes, this is the fail-safe, will look into it.